### PR TITLE
Beacd 110 Toggle for Emac screens

### PR DESCRIPTION
**Pull request by liampan**
Added a toggle to change Urls to emac screens,
Updated partials to use toggle.

### CODE REVIEW

File: app/controllers/helpers/AccountSummaryHelper.scala [hunk 1]
```diff
@@ -84,7 +85,7 @@ class AccountSummaryHelper @Inject()(appConfig: FrontendAppConfig,
   private def buildVatVarsSection(vatDecEnrolment: VatDecEnrolment, vatVarEnrolment: VatEnrolment
                                  )(implicit request: AuthenticatedRequest[_]) : Option[Html] ={
     vatVarEnrolment match {
-      case x: VatEnrolment if !x.enrolled  => Some(vat_var_prompt_to_enrol(appConfig,vatDecEnrolment))
+      case x: VatEnrolment if !x.enrolled  => Some(vat_var_prompt_to_enrol(new EmacUrlBuilder(appConfig), vatDecEnrolment))
```
Comment (kurtlogan) [inline app/controllers/helpers/AccountSummaryHelper.scala]: Use dependency injection for EmacUrlBuilder

File: app/controllers/vat/VatEnrolAddressTabController.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+class VatEnrolAddressTabController @Inject()(
+  override val messagesApi: MessagesApi,
+  authenticate: AuthAction,
+  serviceInfo: ServiceInfoAction,
+  appConfig: FrontendAppConfig
+) extends FrontendController with I18nSupport {
+
+  def onPageLoad(): Action[AnyContent] = authenticate.async {
+    implicit request =>
+      Future.successful(Ok(vat_enrol_address_tab(new EmacUrlBuilder(appConfig), request.vatDecEnrolment)))
```
Comment (kurtlogan) [inline app/controllers/vat/VatEnrolAddressTabController.scala]: Use dependency injection for EmacUrlBuilder

File: app/controllers/vat/VatEnrolBankTabController.scala [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+class VatEnrolBankTabController @Inject()(
+ override val messagesApi: MessagesApi,
+ authenticate: AuthAction,
+ serviceInfo: ServiceInfoAction,
+ appConfig: FrontendAppConfig
+) extends FrontendController with I18nSupport {
+
+  def onPageLoad(): Action[AnyContent] = authenticate.async {
+    implicit request =>
+      Future.successful(Ok(vat_enrol_bank_tab(new EmacUrlBuilder(appConfig), request.vatDecEnrolment)))
```
Comment (kurtlogan) [inline app/controllers/vat/VatEnrolBankTabController.scala]: Use dependency injection for EmacUrlBuilder

File: test/utils/PortalUrlBuilderSpec.scala [hunk 4]
```diff
@@ -17,9 +17,14 @@
 package utils
 
 import base.SpecBase
+import config.FrontendAppConfig
```
Comment (kurtlogan) [inline test/utils/PortalUrlBuilderSpec.scala]: There are no meaningful changes to this file, undo all changes to this file

File: test/views/partials/vat/VatEnrolAddressTabViewSpec.scala [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import play.twirl.api.HtmlFormat
+import uk.gov.hmrc.domain.Vrn
+import utils.EmacUrlBuilder
+import views.behaviours.ViewBehaviours
+import views.html.partials.vat.vat_enrol_address_tab
+
+class VatEnrolAddressTabViewSpec extends ViewBehaviours with MockitoSugar {
+
+  val vatDecEnrolment = VatDecEnrolment(Vrn("a-users-vrn"), isActivated = true)
+
```
Comment (kurtlogan) [inline test/views/partials/vat/VatEnrolAddressTabViewSpec.scala]: Remove extra newline

File: test/views/partials/vat/VatEnrolBankTabViewSpec.scala [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import models.VatDecEnrolment
+import org.scalatest.mockito.MockitoSugar
+import org.mockito.Mockito._
+import play.twirl.api.HtmlFormat
+import uk.gov.hmrc.domain.Vrn
+import utils.EmacUrlBuilder
+import views.behaviours.ViewBehaviours
+import views.html.partials.vat.vat_enrol_bank_tab
+
+
```
Comment (kurtlogan) [inline test/views/partials/vat/VatEnrolBankTabViewSpec.scala]: remove extra newline

File: app/config/FrontendAppConfig.scala [hunk 7]
```diff
@@ -64,8 +64,10 @@ class FrontendAppConfig @Inject() (override val runModeConfiguration: Configurat
   def getUrl(key: String): String = loadConfig(s"urls.$key")
   def getGovUrl(key: String): String = loadConfig(s"urls.external.govuk.$key")
   def getBusinessAccountUrl(key: String): String = businessAccountHost + loadConfig(s"urls.business-account.$key")
-  def getPortalUrl(key: String)(vatEnrolment: Option[VatEnrolment])(implicit request: Request[_]): String =
+
+  def getPortalUrl(key: String)(vatEnrolment: Option[VatEnrolment])(implicit request: Request[_]): String = {
```
Comment (kurtlogan) [inline app/config/FrontendAppConfig.scala]: Remove `{` and `}` from this method

File: app/controllers/vat/VatEnrolAddressTabController.scala [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+import scala.concurrent.Future
+
+class VatEnrolAddressTabController @Inject()(
+  override val messagesApi: MessagesApi,
+  authenticate: AuthAction,
+  serviceInfo: ServiceInfoAction,
+  appConfig: FrontendAppConfig
+) extends FrontendController with I18nSupport {
+
+  def onPageLoad(): Action[AnyContent] = authenticate.async {
```
Comment (kurtlogan) [inline app/controllers/vat/VatEnrolAddressTabController.scala]: Remove `.aync` and `Future.successful` from this method. This method performs a synchronous  action so wrapping it in a future is redundant

File: app/controllers/vat/VatEnrolBankTabController.scala [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+import scala.concurrent.Future
+
+class VatEnrolBankTabController @Inject()(
+ override val messagesApi: MessagesApi,
+ authenticate: AuthAction,
+ serviceInfo: ServiceInfoAction,
+ appConfig: FrontendAppConfig
+) extends FrontendController with I18nSupport {
+
+  def onPageLoad(): Action[AnyContent] = authenticate.async {
```
Comment (kurtlogan) [inline app/controllers/vat/VatEnrolBankTabController.scala]: Remove `.async` and `Future.successful` as this action is syncronous

File: app/utils/EmacUrlBuilder.scala [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+import config.FrontendAppConfig
+import models.{VatDecEnrolment, VatEnrolment}
+import play.api.mvc.Request
+import uk.gov.hmrc.domain.Vrn
+
+class EmacUrlBuilder@Inject()(appConfig: FrontendAppConfig) {
+
+  def getRequestAccessUrl(enrolmentKey: String)(vatEnrolment: Option[VatEnrolment])(implicit request: Request[_]): String = {
+
+    val vatDecEnrolment = VatDecEnrolment(Vrn("a-users-vrn"), isActivated = true)
```
Comment (kurtlogan) [inline app/utils/EmacUrlBuilder.scala]: Why is this here? This looks like this should be removed and `vatEnrolment` should be passed to `getPortalUrl`

File: app/utils/EmacUrlBuilder.scala [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+    if (appConfig.useEmacVatEnrolment)
+      s"/enrolment-management-frontend/HMCE-VATVAR-ORG/request-access-tax-scheme?continue=%2Fbusiness-account"
```
Comment (kurtlogan) [inline app/utils/EmacUrlBuilder.scala]: Move this to the `conf` file

File: app/utils/EmacUrlBuilder.scala [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+import javax.inject.Inject
+
+import config.FrontendAppConfig
+import models.{VatDecEnrolment, VatEnrolment}
+import play.api.mvc.Request
+import uk.gov.hmrc.domain.Vrn
+
+class EmacUrlBuilder@Inject()(appConfig: FrontendAppConfig) {
+
+  def getRequestAccessUrl(enrolmentKey: String)(vatEnrolment: Option[VatEnrolment])(implicit request: Request[_]): String = {
```
Comment (kurtlogan) [inline app/utils/EmacUrlBuilder.scala]: Everywhere this method is called the `vatEnrolment` is wrapped in a `Some`. It would be better to not make this optional and only wrap it in a `Some` when it is needed

File: test/views/partials/vat/VatEnrolBankTabViewSpec.scala [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+  def createView: () => HtmlFormat.Appendable =
+    () => vat_enrol_bank_tab(new EmacUrlBuilder(frontendAppConfig), vatDecEnrolment)(fakeRequest, messages)
+
+  "Vat enrol bank tab partial" should {
+    "display correct content" in {
+      asDocument(createView()).getElementById("vat-enrol-bank-tab").text() must include("You can't change your repayment account yet.")
+    }
+
+    val enrolLink = asDocument(createView()).getElementById("vat-activate-or-enrol-details-bank")
```
Comment (kurtlogan) [inline test/views/partials/vat/VatEnrolBankTabViewSpec.scala]: Add some whitespace after this line to break it up slightly

Comment (kurtlogan) [inline test/views/partials/vat/VatEnrolBankTabViewSpec.scala]: Add some whitespace after this line to break it up slightly

File: app/controllers/vat/VatEnrolBankTabController.scala [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import utils.EmacUrlBuilder
+import views.html.partials.vat.vat_enrol_bank_tab
+
+import scala.concurrent.Future
+
+class VatEnrolBankTabController @Inject()(
+ override val messagesApi: MessagesApi,
+ authenticate: AuthAction,
+ serviceInfo: ServiceInfoAction,
```
Comment (kurtlogan) [inline app/controllers/vat/VatEnrolBankTabController.scala]: `serviceInfo: ServiceInfoAction`  is not used

File: app/controllers/vat/VatEnrolAddressTabController.scala [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import utils.EmacUrlBuilder
+import views.html.partials.vat.vat_enrol_address_tab
+
+import scala.concurrent.Future
+
+class VatEnrolAddressTabController @Inject()(
+  override val messagesApi: MessagesApi,
+  authenticate: AuthAction,
+  serviceInfo: ServiceInfoAction,
```
Comment (kurtlogan) [inline app/controllers/vat/VatEnrolAddressTabController.scala]: `serviceInfo: ServiceInfoAction`  is not used

**Comment by liampan** [comment]
@kurtlogan please review changes

File: app/config/FrontendAppConfig.scala [hunk 16]
```diff
@@ -77,4 +78,7 @@ class FrontendAppConfig @Inject() (override val runModeConfiguration: Configurat
 
   lazy val changesToVat = runModeConfiguration.getBoolean("microservice.services.features.changes-to-vat").getOrElse(false)
   lazy val changesToVatUrl = runModeConfiguration.getString("urls.external.govuk.changesToVat").getOrElse("")
+
+  lazy val useEmacVatEnrolment = runModeConfiguration.getBoolean("microservice.services.features.emac-vat-enrolment").getOrElse(false)
+  lazy val EmacVatEnrolmentUrl = loadConfig("urls.external.emac.enrol")
```
Comment (kurtlogan) [inline app/config/FrontendAppConfig.scala]: Lower case `E` here

File: app/controllers/vat/VatEnrolBankTabController.scala [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+
+class VatEnrolBankTabController @Inject()(
+ override val messagesApi: MessagesApi,
+ authenticate: AuthAction,
+ appConfig: FrontendAppConfig
+) extends FrontendController with I18nSupport {
+
+  def onPageLoad(): Action[AnyContent] = authenticate{
+    implicit request =>
+      Ok(vat_enrol_bank_tab(new EmacUrlBuilder(appConfig), request.vatDecEnrolment))
```
Comment (kurtlogan) [inline app/controllers/vat/VatEnrolBankTabController.scala]: Use dependency injection here

File: test/views/partials/vat/VatEnrolBankTabViewSpec.scala [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+import utils.EmacUrlBuilder
+import views.behaviours.ViewBehaviours
+import views.html.partials.vat.vat_enrol_bank_tab
+
+class VatEnrolBankTabViewSpec extends ViewBehaviours with MockitoSugar {
+
+  val vatDecEnrolment = VatDecEnrolment(Vrn("a-users-vrn"), isActivated = true)
+
+  def createView: () => HtmlFormat.Appendable =
+    () => vat_enrol_bank_tab(new EmacUrlBuilder(frontendAppConfig), vatDecEnrolment)(fakeRequest, messages)
```
Comment (kurtlogan) [inline test/views/partials/vat/VatEnrolBankTabViewSpec.scala]: use `emacUrlBuilder`

File: test/views/partials/vat/VatEnrolAddressTabViewSpec.scala [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+  def createView: () => HtmlFormat.Appendable =
+    () => vat_enrol_address_tab(new EmacUrlBuilder(frontendAppConfig), vatDecEnrolment)(fakeRequest, messages)
```
Comment (kurtlogan) [inline test/views/partials/vat/VatEnrolAddressTabViewSpec.scala]: use `emacUrlBuilder`

File: test/controllers/vat/VatEnrolBankTabControllerSpec.scala [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+import utils.EmacUrlBuilder
+import views.html.partials.vat.vat_enrol_bank_tab
+
+class VatEnrolBankTabControllerSpec extends ControllerSpecBase with MockitoSugar with BeforeAndAfterEach {
+
+  def controller() = new VatEnrolBankTabController(messagesApi, FakeAuthAction, frontendAppConfig)
+
+  val vatDecEnrolment = VatDecEnrolment(Vrn("a-users-vrn"), isActivated = true)
+
+  def viewAsString() = vat_enrol_bank_tab(new EmacUrlBuilder(frontendAppConfig), vatDecEnrolment)(fakeRequest, messages).toString
```
Comment (kurtlogan) [inline test/controllers/vat/VatEnrolBankTabControllerSpec.scala]: use `emacUrlBuilder`

File: test/controllers/vat/VatEnrolAddressTabControllerSpec.scala [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+import views.html.partials.vat.vat_enrol_address_tab
+
+
+class VatEnrolAddressTabControllerSpec extends ControllerSpecBase with MockitoSugar {
+
+  def controller() = new VatEnrolAddressTabController(messagesApi, FakeAuthAction, frontendAppConfig, emacUrlBuilder)
+
+  val vatDecEnrolment = VatDecEnrolment(Vrn("a-users-vrn"), isActivated = true)
+
+  def viewAsString() = vat_enrol_address_tab(new EmacUrlBuilder(frontendAppConfig), vatDecEnrolment)(fakeRequest, messages).toString
```
Comment (kurtlogan) [inline test/controllers/vat/VatEnrolAddressTabControllerSpec.scala]: use `emacUrlBuilder`
